clouds/aws/aws_utils.py

```diff
@@ -139,3 +139,34 @@ def delete_bucket(bucket_name: str, boto_client) -> None:
         f"Bucket {bucket_name} not deleted",
         json.dumps(response, defualt=str, indent=4),
     )
+
+
+def aws_region_names():
+    """
+    Lists AWS regions
+
+    Returns:
+        list: list of AWS regions name
+
+    """
+    regions = ec2_client().describe_regions()["Regions"]
+    return [region["RegionName"] for region in regions]
+
+
+def get_least_crowded_aws_vpc_region(region_list):
+    """
+    Selects region with the least number of VPCs.
+
+    Args:
+        region_list: list of region names
+
+    Returns:
+        str: region name with have the least number of VPCs
+
+    """
+    region, vpcs = 0, 0
```

Comment:
But, we may end without any region, please set here:
`region, vpcs = region_list[0], 0` to make sure that we will always return a region.

Reply:
also raise and log if `region_list` is empty before do this.

Reply:
updated the code to check the region_list